tools/src/wyvern/tools/typedAST/core/declarations/DefDeclaration.java

```diff
@@ -235,29 +242,62 @@ public Environment extendName(Environment env, Environment against) {
 	@Override
 	public DeclType genILType(GenContext ctx) {
 		List<FormalArg> args = new LinkedList<FormalArg>();
-		for (NameBinding b : argNames) {
+
+        ctx = this.serializeArguments(args, ctx);
+
+		DefDeclType ret = new DefDeclType(getName(), getResultILType(ctx), args);
+		return ret;
+	}
+
+    private GenContext serializeArguments(List<FormalArg> args, GenContext ctx) {
+        if(isGeneric()) {
+            for(String s : this.generics) {
+                ValueType type = this.genericStructuralType();
+                args.add(new FormalArg(GENERIC_PREFIX, type));
```

Comment:
For a function with N parameters, you are adding N arguments that all have the same name.  This doesn't make sense.  Instead of calling each one GENERIC_PREFIX, I suggest you call each one GENERIC_PREFIX+s